### ISSUE TITLE
Random Lesson Order

### ISSUE THREAD

**Issue by Phrim1** [davidsansome/tsurukame#201]
It would be nice to be able to sort Lessons by random instead of some combination of Radical/Kanji/Vocab.  This is implemented on the web with the option "Lesson Ordering: Shuffled".

**Comment by zebala** [comment]
+1 to this. Recently I've come to hate the order in which WK gives the items, since it groups vocab using the same kanji together. This backfires in reviews as you may fail one but then remember the other ones even though you didn't. IMO it's better to sprinkle them. Also makes reviews and lessons a bit more interesting, and you don't get lots of similar items together.

**Comment by turtlemaster19** [comment]
I was thinking we could do something like the below image, and have the cells be able to move between the two sections. Obviously, shuffling just vocabulary would do nothing, but the idea is that the cells would be able to move between both.

![UI Mockup](https://user-images.githubusercontent.com/46784000/75450389-535d0d00-5934-11ea-96d9-bc0c3e7fd34c.png)

**Comment by zebala** [comment]
Looks pretty nice! Wouldn't the vocab shuffling still change the order of the vocab items? Assuming we don't have any pending radicals or kanji.

**Comment by turtlemaster19** [comment]
@zebala That might actually be a good idea if that were implemented. We would then just completely randomise anything in the shuffle group, and keep our current order outside of it...

**Comment by Phrim1** [comment]
That looks awesome!

**Comment by zebala** [comment]
I wonder if I wanted to do some vocab lessons from the previous level shuffled could be handled. Like in WK "Ascending level then shuffled". Basically if there is vocab spanning multiple levels.

**Comment by turtlemaster19** [comment]
@zebala Good point. I think if we implement shuffling, level should be taken care of first. For example, if prioritize current level is on, that would take precedence over the other shuffling things.
